app_dart/lib/src/request_handlers/scheduler/vacuum_stale_tasks.dart

```diff
@@ -32,8 +32,9 @@ class VacuumStaleTasks extends RequestHandler<Body> {
   /// For testing, can be used to inject a deterministic time.
   final DateTime? nowValue;
 
-  /// Tasks that are in progress for this duration will be reset.
-  static const Duration kTimeoutLimit = Duration(hours: 3);
+  /// Tasks that are in progress without a build for this duration will be
+  /// reset.
+  static const Duration kTimeoutLimit = Duration(minutes: 30);
```

Comment:
This seems a little aggressive, does this need to be lowered?

Reply:
The code only triggers if the build ID is `null`. Maybe I'm wrong about the protocol here, but I thought we would expect to get a build ID back from LUCI pretty much right away, and if that's the case than 30 minutes is a very long time.

I can put it back to 3 hours, but that means that when this case happens where an entire block of 50 backfill builds never gets scheduled, it'll be 3-4 hours before we even start to recover those columns, which seemed undesirable. The 3 hour timeout was from the original code that didn't check for `null` builds, which meant that it would restart any job older than 3 hours that was in progress *even if we knew it was running*, which I assume is why it was so high.

Are there expected cases where we wouldn't get a build number from LUCI within 30 minutes?

Reply:
is the build ID filled in if it's pending but not started? If so, then 30 minutes SGTM.

Reply:
I asked LUCI folks about the potential max delay here, and it turns out that we should actually be getting the build ID back *immediately* when we do the RPC, we don't actually have to wait for the pubsub like we did with v1. I started to look and it didn't seem that we were actually using that information, at least for backfill.

So the build ID exists ~immediately, but I don't know when we get it via pubsub. I'll adjust this back to three hours for now to make the change more conservative, land it for incremental tree health improvement, and then start looking into using the RPC response data so it's more robust.

Reply:
Cool, SGTM